x/crosschain/keeper/msg_add_to_intx_tracker.go

```diff
@@ -0,0 +1,93 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// TODO https://github.com/zeta-chain/node/issues/1269
+func (k msgServer) AddToInTxTracker(goCtx context.Context, msg *types.MsgAddToInTxTracker) (*types.MsgAddToInTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observertypes.ErrSupportedChains
+	}
+
+	adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observertypes.Policy_Type_group1)
+	isAdmin := msg.Creator == adminPolicyAccount
+
+	isObserver, err := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+	if err != nil {
+		ctx.Logger().Error("Error while checking if the account is an observer", err)
+	}
+	isProven := false
+	if !(isAdmin || isObserver) && msg.Proof != nil {
+		txx, err := k.VerifyProof(ctx, msg.Proof, msg.BlockHash, msg.TxIndex, msg.ChainId)
+		if err != nil {
+			return nil, types.ErrCannotVerifyProof.Wrapf(err.Error())
+		}
+		err = k.VerifyInTxTrackerProof(ctx, txx, msg.ChainId, msg.CoinType)
+		if err != nil {
+			return nil, types.ErrCannotVerifyProof.Wrapf(err.Error())
+		}
+		isProven = true
+	}
+
+	// Sender needs to be either the admin policy account or an observer
+	if !(isAdmin || isObserver || isProven) {
```

Comment:
This line allows an observer reporting an arbitrary inTx hash to create the tracker (line 49) without proof or majority vote?

Reply:
yes , that is the existing logic already